packages/app/src/vis-packs/core/hooks.ts

```diff
@@ -78,10 +78,10 @@ export function useDatasetValues<D extends Dataset<ArrayShape | ScalarShape>>(
   });
 }
 
-export function useBaseArray<T, U extends T[] | TypedArray | undefined>(
-  value: U,
+export function useBaseArray<T extends unknown[] | TypedArray | undefined>(
+  value: T,
```

Comment:
This is the main change throughout. Before, you could do `useBaseArray<string, string[]>(...)` to effectively type-cast the return value to `NdArray<string>` -- i.e. like doing `useBaseArray(...) as NdArray<string>` but less explicitly, which was bad.

Reply:
One less generic is a win for me :+1: